pkg/storage/engine/temp_engine.go

```diff
@@ -62,52 +93,102 @@ type tempEngine struct {
 
 func (e *tempEngine) Close() {
 	e.RocksDB.Close()
+	// We could call cleanupTempStoreDirs here, but since we prefer faster
+	// shutdowns (whereas node start-up allow more time for us to cleanup
+	// asynchronously) we simply remove our current temporary store.
 	dir := e.RocksDB.cfg.Dir
 	if dir == "" {
 		return
 	}
 	if err := os.RemoveAll(dir); err != nil {
-		log.Errorf(context.TODO(), "could not remove rocksdb dir: %v", err)
+		log.Errorf(context.TODO(), "could not remove temporary store directory: %v", err.Error())
 	}
 }
 
-// wg is allowed to be nil, if the caller does not want to wait on the cleanup.
-func cleanupTempStorageDirs(ctx context.Context, path string, wg *sync.WaitGroup) error {
-	// Removing existing contents might be slow. Instead we rename it to a new
-	// name, and spawn a goroutine to clean it up asynchronously.
-	if err := os.MkdirAll(path, 0755); err != nil {
-		return err
-	}
-	deletionDir, err := ioutil.TempDir(path, "TO-DELETE-")
+// logTempStoreDir records the location of the temporary directory in a log
+// file for cleanup on startup if necessary.
+func logTempStoreDir(storePath string, tempPath string) error {
+	return logTempStoreDirs(storePath, append([]byte(tempPath), '\n'))
+}
+
+// logTempStoreDirs records tempPaths (list of temporary directory paths) to a log file
+// file for cleanup on startup if necessary.
+// tempPaths is a \n separated and ending list of paths to the temporary store directories.
+func logTempStoreDirs(storePath string, tempPaths []byte) error {
```

Comment:
This allows us to (concurrently) write to the log file for:
1. New temporary directory https://github.com/cockroachdb/cockroach/pull/18544/files#diff-5ae65fe6be0e091bd0ee5d29d87e66f6R83
2. Directories that could not be cleaned up during startup https://github.com/cockroachdb/cockroach/pull/18544/files#diff-5ae65fe6be0e091bd0ee5d29d87e66f6R191

Again, would love some feedback on this approach!